src/Utilities/Compiler/Options/AnalyzerOptionsExtensions.cs

```diff
@@ -272,41 +273,79 @@ static bool TryParseValue(string value, out string result)
             CancellationToken cancellationToken)
             => options.GetSymbolNamesWithValueOption<Unit>(EditorConfigOptionNames.AdditionalStringFormattingMethods, rule, tree, compilation, cancellationToken, namePrefixOpt: "M:");
 
-        public static SymbolNamesWithValueOption<Unit> GetExcludedSymbolNamesWithValueOption(
-            this AnalyzerOptions options,
-            DiagnosticDescriptor rule,
-            ISymbol symbol,
-            Compilation compilation,
-            CancellationToken cancellationToken)
-        => TryGetSyntaxTreeForOption(symbol, out var tree)
-            ? options.GetExcludedSymbolNamesWithValueOption(rule, tree, compilation, cancellationToken)
-            : SymbolNamesWithValueOption<Unit>.Empty;
 
-        public static SymbolNamesWithValueOption<Unit> GetExcludedSymbolNamesWithValueOption(
-            this AnalyzerOptions options,
+        /// <summary>
+        /// Returns true if the given source symbol has been configured to be excluded from analysis by options.
+        /// </summary>
+        public static bool IsConfiguredToSkipAnalysis(
+            this ISymbol symbol,
```

Comment:
It seems weird to have extension method with symbol as the 'this' parameter in this file. Instead a better approach would be to revert the change to `ISymbolExtensions` and define new extension methods in this file with `this AnalyzerOptions options` which just calls back into the existing extension method on `ISymbol`.

Reply:
That's a good remark. We do have a couple of helpers in this class that take a symbol as extra argument so I will try a different signature. If you still think that's no good I will refactor and use your suggested solution.